src/components/dropdown/dropdown.js

```diff
@@ -1,41 +1,104 @@
-import { html, LitElement } from '@polymer/lit-element'; 
+import { html, LitElement } from '@polymer/lit-element';
+import css from './dropdown.css';
 
 class DropdownComponent extends LitElement {
-
   static get properties() {
     return {
-      optionSelectedCallback: {
-        type: Function
-      },
       label: {
         type: String
       },
       options: {
         type: Array
+      },
+      selected: {
+        type: String
       }
     };
   }
 
-  /* eslint-disable indent */
+  handleEvent(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    let selected = '';
+
+    if (e.target.htmlFor) {
+      selected = e.target.parentNode;
+      if (e.target.htmlFor !== 'option_def') {
+        this.setSelected(this.shadowRoot.querySelector('#' + e.target.htmlFor).value);
+        selected.classList.add('selection');
+      }
+    } else {
+      selected = e.target;
+    }
+    this.handleDropdown(selected);
+    selected.classList.toggle('expanded');
+  }
+
+  setSelected(selected) {
+    let idx = this.options.findIndex(({ value }) => {
+      return value === selected;
+    });
+
+    let previousCheck = this.options.findIndex(({ checked }) => {
+      return checked;
+    });
+
+    if (previousCheck > -1) {
+      this.options[previousCheck].checked = false;
+    }
+
+    if (idx > -1) {
+      this.options[idx].checked = true;
+      this.selected = this.options[idx].value;
+      this.optionSelectedCallback(this.selected, idx);
+    }
+    this.dropdown.scrollTop = 0;
+  }
+
+  handleDropdown(drop) {
```

Comment:
A couple name changes here might improve the readability of the code a bit more:
- Consider calling it `setSelectedDropdownOption`?
- It might be a bit more explicit to call it `selectedOption`, as opposed to `drop`